### ISSUE TITLE
Bug: First apply fails due to an apply time dependency

### ISSUE THREAD

**Issue by jaredfholgate** [Azure/terraform-azurerm-avm-ptn-alz#8]
This line is causing a dependency on an apply time value for plan. https://github.com/Azure/terraform-azurerm-avm-ptn-alz/blob/233be00b6a644f6b6ff49f1bb9c233fde5f4cf01/main.tf#L209
 
I think the reason is that it needs the `parent_id` of the parent management group, which is not known until after the first apply.
 
I have temporarily updated to use a different map for now, but it will not be able to determine whether role assignments are required based on the archetypes, so not sure the best way to handle this delay function. The workaround will be fine, it will just cause a delay when one may not be needed.
 
I am happy to raise a PR for this workaround to ensure that consumers of the module are not impacted though if someone can grant me access to create a branch.

Suggested interim update:

From this:

```hcl
resource "time_sleep" "before_policy_role_assignments" {
  count            = local.policy_role_assignments != {} ? 1 : 0
  create_duration  = var.delays.before_policy_role_assignments.create
  destroy_duration = var.delays.before_policy_role_assignments.destroy

  triggers = {
    policy_assignments = jsonencode(azurerm_management_group_policy_assignment.this)
  }

  depends_on = [azurerm_management_group_policy_assignment.this]
}
```

To this: 

```hcl
resource "time_sleep" "before_policy_role_assignments" {
  count            = local.alz_policy_assignments_decoded != {} ? 1 : 0
  create_duration  = var.delays.before_policy_role_assignments.create
  destroy_duration = var.delays.before_policy_role_assignments.destroy

  triggers = {
    policy_assignments = jsonencode(azurerm_management_group_policy_assignment.this)
  }

  depends_on = [azurerm_management_group_policy_assignment.this]
}
```

**Comment by matt-FFFFFF** [comment]
Hi @jaredfholgate 

Thank you for this. I think this is a sensible suggestion.

**Comment by jaredfholgate** [comment]
Released in v0.3.2
